core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -446,6 +447,64 @@ def validate_funding_profiles_funding_secured_not_null(
         ]
 
 
+def validate_psi_funding_not_negative(
+    workbook: dict[str, pd.DataFrame]
+) -> list["TownsFundRoundFourValidationFailure"] | None:
+    """
+    Validates that Private Sector Funding amounts are not negative.
+
+    This function checks the 'Private Investments' sheet for rows where either
+    'Private Sector Funding Required' or 'Private Sector Funding Secured' is negative.
+    If invalid rows are found, it returns a list of validation failures.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                     DataFrames representing each sheet in the Round 4 submission.
+    :return: ValidationErrors or None if no errors are found.
+    """
+    psi_df = workbook["Private Investments"]
+    psi_df["Private Sector Funding Required"] = pd.to_numeric(
+        psi_df["Private Sector Funding Required"], errors="coerce"
+    )
+    invalid_mask = psi_df["Private Sector Funding Required"] < 0
+
+    invalid_psi_rows = psi_df.loc[invalid_mask]
+    errors = []
+    if len(invalid_psi_rows) > 0:
+        invalid_indexes = invalid_psi_rows.index.tolist()
+        errors.extend(
+            [
+                TownsFundRoundFourValidationFailure(
+                    sheet="Private Investments",
+                    section="Private Sector Funding Required",
+                    column="Private Sector Funding Required",
+                    message=msgs.NEGATIVE_NUMBER,
+                    row_indexes=invalid_indexes,
+                )
+            ]
+        )
+
+    psi_df["Private Sector Funding Secured"] = pd.to_numeric(psi_df["Private Sector Funding Secured"], errors="coerce")
+    invalid_mask = psi_df["Private Sector Funding Secured"] < 0
+
+    invalid_psi_rows = psi_df.loc[invalid_mask]
+    if len(invalid_psi_rows) > 0:
+        invalid_indexes = invalid_psi_rows.index.tolist()
+        errors.extend(
+            [
+                TownsFundRoundFourValidationFailure(
+                    sheet="Private Investments",
+                    section="Private Sector Investment",
+                    column="Private Sector Funding Secured",
+                    message=msgs.NEGATIVE_NUMBER,
+                    row_indexes=invalid_indexes,
+                )
+            ]
+        )
```

Comment:
Is it possible to instead define a tuple of columns to check and look over it to reduce code repetition 

e.g.
```
cols_to_check = ("Private Sector Funding Required", "Private Sector Funding Secured")

for col in cols to check:
    # do the check
```

Reply:
I've refactored the entire function now so that it's a lot leaner

Reply:
looks good